src/Public/OpenApi.ps1

```diff
@@ -66,48 +102,163 @@ function Enable-PodeOpenApi {
+        }
+        if ($WebEvent.path -ilike '*.json') {
+            if ($format) {
+                Show-PodeErrorPage -Code 400 -ContentType 'text/html' -Description 'Format query not valid when the file extension is used'
+                return
+            }
+            $format = 'json'
+        } elseif ($WebEvent.path -ilike '*.yaml') {
+            if ($format) {
+                Show-PodeErrorPage -Code 400 -ContentType 'text/html' -Description 'Format query not valid when the file extension is used'
+                return
+            }
+            $format = 'yaml'
+        } elseif (!$format) {
+            $format = $meta.MarkupLanguage.ToLower()
+        } elseif (@('yaml', 'json') -inotcontains $format) {
+            Show-PodeErrorPage -Code 400 -ContentType 'text/html' -Description "Format $format not valid"
+            return
+        }
+
+        if (($mode -ieq 'download')  ) {
+            # Set-PodeResponseAttachment -Path
+            Add-PodeHeader -Name 'Content-Disposition' -Value "attachment; filename=openapi.$format"
+        }
 
         # generate the openapi definition
         $def = Get-PodeOpenApiDefinitionInternal `
-            -Title $PodeContext.Server.OpenAPI.Title `
-            -Version $meta.Version `
-            -Description $meta.Description `
-            -RouteFilter $meta.RouteFilter `
             -Protocol $WebEvent.Endpoint.Protocol `
             -Address $WebEvent.Endpoint.Address `
             -EndpointName $WebEvent.Endpoint.Name `
-            -RestrictRoutes:$strict
+            -MetaInfo $meta
 
         # write the openapi definition
-        Write-PodeJsonResponse -Value $def -Depth 20
+        if ($format -ieq 'yaml') {
+            if ($mode -ieq 'view') {
+                Write-PodeTextResponse -Value (ConvertTo-PodeYaml -InputObject $def -depth 10) -ContentType 'text/x-yaml; charset=utf-8'
+            } else {
+                Write-PodeYamlResponse -Value $def -depth 10
+            }
+        } else {
+            Write-PodeJsonResponse -Value $def -depth 10 -NoCompress:$meta.NoCompress
+        }
+    }
+
+    # add the OpenAPI route
+    Add-PodeRoute -Method Get -Path $Path -ArgumentList $meta -Middleware $Middleware -ScriptBlock $openApiCreationScriptBlock
+    Add-PodeRoute -Method Get -Path "$Path.json" -ArgumentList $meta -Middleware $Middleware -ScriptBlock $openApiCreationScriptBlock
+    Add-PodeRoute -Method Get -Path "$Path.yaml" -ArgumentList $meta -Middleware $Middleware -ScriptBlock $openApiCreationScriptBlock
+
+}
+
+<#
+.SYNOPSIS
+Creates an OpenAPI Server property.
+
+.DESCRIPTION
+Creates an OpenAPI Server property.
+
+.PARAMETER Url
+Server or path to local server.
+
+.PARAMETER Description
+Description of the server.
+
+.EXAMPLE
+Add-PodeOpenApiServerEndpoint -Url 'https://myserver.io/api' -Description 'My test server'
+
+.EXAMPLE
+Add-PodeOpenApiServerEndpoint -Url '/api' -Description 'My local server'
+#>
+function Add-PodeOpenApiServerEndpoint {
```

Comment:
Just to match the other functions, outside of enabling OpenAPI and getting the definition, this should be renamed to `Add-PodeOAServerEndpoint`

Reply:
I agree